DeePVP
==============

### Introduction

DeePVP is a tool for the identification and classification of phage virion protein (PVP) using deep learning approach. The main module of DeePVP aims to discriminate PVPs from non-PVPs within a phage genome, while the extended module of DeePVP can further classify predicted PVPs into the ten major classes of PVPs, namely head-tail joining, collar, tail sheath, tail fiber, portal, minor tail, major tail, baseplate, minor capsid, and major capsid, which are the dominant categories in PVP. DeePVP takes a phage protein sequence file in "fasta" format as input, and outputs a tabular file. The program is optimized in both a virtual machine with GUI and a docker.

![151703377-b3153bab-911e-49f6-b16d-9b43fdef9eed](https://user-images.githubusercontent.com/107048586/172389231-10ce628c-e167-4f22-be2e-235128063fe3.png)

Manuals
-------

[DeePVP manual.pdf] (https://github.com/fangzcbio/DeePVP/blob/main/DeePVP%20manual.pdf)



